### PR TITLE
Fix for issue #2840

### PR DESCRIPTION
**Pull request by davidreis97**
**Summary**

This is a fix for issue #2840 .

It fixes the issue by comparing the name of the git repo with the entries on the lockfile, instead of comparing the whole git url (which wouldn't match with the version already on the lockfile, since the url would be different).

**Test plan**

We tested with multiple git repositories, including the ones that were given as an example on the issue page.

### CODE REVIEW

**Comment by kaylieEB** [comment]
@davidreis97 Thanks for the PR! can you add a test case for this?

**Comment by davidreis97** [comment]
Hi @kaylieEB , I just added a test case in the last commit.

File: src/cli/commands/install.js [hunk 1]
```diff
@@ -234,14 +237,20 @@ export class Install {
...
-      // can't extract a package name from this
-      if (getExoticResolver(pattern)) {
+      if (GitResolver.isVersion(pattern)) {
+        const pathname = urlParse(pattern).pathname;
+
+        if (pathname) {
+          const name = util.removeSuffix(pathname.split('/')[2], '.git');
+          excludeNames.push(name);
+        }
+      } else if (getExoticResolver(pattern)) {
```
Comment (kaylieEB) [inline src/cli/commands/install.js]: Can you instead try using [guessName util](https://github.com/yarnpkg/yarn/blob/8918da15c36e793c8f5ed7be24197a7b5afc89d4/src/util/guess-name.js), so something like
```
if (getExoticResolver(pattern)) {
   excludeNames.push(guessName(pattern))
}
```
This way we avoid git-specific logic in the install command file and other exotic resolvers and hosted git resolvers can't take advantage of this fix too :)

File: __tests__/commands/add.js [hunk 2]
```diff
@@ -609,6 +609,27 @@ test.concurrent('upgrade scenario 2 (with sub dependencies)', (): Promise<void>
   });
 });
 
+test.concurrent('install another fork of an existing package', (): Promise<void> => {
+  // When installing a package with the same name as an existing one but from a different repo,
+  // the old one should be replaced with the new one in the lock file.
+  const firstSource = 'https://github.com/davidreis97/example-yarn-package.git';
+  const secondSource = 'https://github.com/yarnpkg/example-yarn-package.git';
```
Comment (kaylieEB) [inline __tests__/commands/add.js]: To avoid making requests in the ci, can you add the tarball cache to `__tests__/fixtures/request-cache/GET/github.com`

**Pull request merged**